packages/payload/src/queues/operations/runAllJobs/index.ts

```diff
@@ -0,0 +1,226 @@
+    return {
+      noJobsRemaining: true,
+    }
+  }
+
+  if (jobsQuery?.docs?.length) {
+    req.payload.logger.info(
+      `${jobsQuery.docs.length} job(s) found, ${newJobs.length} of which are new. Running ${limit} job(s).`,
+    )
+  }
+
+  const jobPromises = jobsQuery.docs.map(async (job) => {
+    if (!job.workflowSlug && !job.taskSlug) {
+      throw new Error('Job must have either a workflowSlug or a taskSlug')
+    }
+
+    const workflowConfig: WorkflowConfig<WorkflowTypes> = job.workflowSlug
+      ? req.payload.config.jobs.workflows.find(({ slug }) => slug === job.workflowSlug)
+      : {
+          slug: 'singleTask',
+          handler: async ({ job, runTask }) => {
+            await runTask({
+              id: '1',
+              input: job.input,
+              task: job.taskSlug as string,
+            })
+          },
+        }
+
+    if (!workflowConfig) {
+      return null // Skip jobs with no workflow configuration
+    }
+
+    const updateJob = getUpdateJobFunction(job, req)
+
+    // the runner will either be passed to the config
+    // OR it will be a path, which we will need to import via eval to avoid
+    // Next.js compiler dynamic import expression errors
+    let workflowHandler: WorkflowHandler<WorkflowTypes> | WorkflowJSON<WorkflowTypes>
+
+    if (
+      typeof workflowConfig.handler === 'function' ||
+      (typeof workflowConfig.handler === 'object' && Array.isArray(workflowConfig.handler))
+    ) {
+      workflowHandler = workflowConfig.handler
+    } else {
+      workflowHandler = await importHandlerPath<typeof workflowHandler>(workflowConfig.handler)
+
+      if (!workflowHandler) {
+        const errorMessage = `Can't find runner while importing with the path ${workflowConfig.handler} in job type ${job.workflowSlug}.`
+        req.payload.logger.error(errorMessage)
+
+        await updateJob({
+          error: {
+            error: errorMessage,
+          },
+          hasError: true,
+          processing: false,
+        })
+
+        return
+      }
+    }
+
+    if (typeof workflowHandler === 'function') {
+      const result = await runJob({
+        job,
+        req,
+        updateJob,
+        workflowConfig,
+        workflowHandler,
+      })
+      return { id: job.id, result }
+    } else {
+      const result = await runJSONJob({
+        job,
+        req,
+        updateJob,
+        workflowConfig,
+        workflowHandler,
+      })
+      return { id: job.id, result }
+    }
+  })
+
+  const resultsArray = await Promise.all(jobPromises)
+  const resultsObject: RunAllJobsResult['jobStatus'] = resultsArray.reduce((acc, cur) => {
+    if (cur !== null) {
+      // Check if there's a valid result to include
+      acc[cur.id] = cur.result
+    }
+    return acc
+  }, {})
+
+  let noJobsRemaining = true
+  for (const jobID in resultsObject) {
+    const jobResult = resultsObject[jobID]
+    if (jobResult.status === 'error') {
+      noJobsRemaining = false // Can be retried
```

Comment:
this is really only showing if any jobs failed and need to retry. we can keep this, but we should have `remainingJobs: 34230` or similar.  

There could still be remaining jobs if you had a limit, but no job failed.